### ISSUE TITLE
Keyboard navigation "sabotaged" by focus-reset functionality

### ISSUE THREAD

**Issue by perqa** [sveltejs/kit#3381]
### Describe the problem

I am trying to implement key navigation in a Sveltekit app. The focused element is indicated by an outline, navigation is done by arrow keys and clicking by Enter or OK key. The problem is that focus is set to the body tag every so often.

In this file: 
.svelte-kit/dev/runtime/internal/start.js
I found these lines:
```
// make it possible to reset focus
document.body.setAttribute('tabindex', '-1');
```

### Describe the proposed solution

The workaround I have implemented now is the following: in `onMount` of `__layout.svelte`, I execute
`document.body.removeAttribute('tabindex');`
but this is an ugly hack. 

Is there currently any better way to handle this? Otherwise I suggest a change to make Sveltekit more key-navigation-friendly. Perhaps a config parameter in svelte.config.js to enable disable tabindex on body? Or a flag in the app to use when switching between mouse and keyboard navigation.

### Importance

would make my life easier

### Additional Information

With the growing demand and requirements for web app accessibility, where keyboard navigation plays a central role, this should be prioritised. In TV applications, keyboard navigation is used almost exclusively (with remote controls).

**Comment by geoffrich** [comment]
This is intentional, and is intended to improve accessibility. `tabindex="-1"` is set on the body element so that it can receive focus. Whenever SvelteKit navigates to a new page (via an `<a>` or by calling `goto()`), it resets focus to the body element. This is to simulate how the browser behaves when doing a full-page reload -- see https://github.com/sveltejs/kit/issues/307 for further explanation.

If you are using [goto](https://kit.svelte.dev/docs#modules-$app-navigation) to navigate, you can set `keepfocus` to true to keep focus on the current element instead of resetting it. However, if you do this, you need to make sure that the current element will still exist so that focus is not lost.

```js
goto('/newpage', { keepfocus: true });
```

If this is not enough to support your use case, please provide some sample code to show what you're trying to do.

**Comment by perqa** [comment]
Thank you! This is basically what I was looking for. I think this should cover my use case, unless there are any other scenarios in which focus is sent to body. I will test.

**Comment by geoffrich** [comment]
Closing this since it sounds like `keepfocus: true` should do what you need. Please reopen if that's not the case.
